### PR TITLE
Add TestMultipleHosts and TestTimeout

### PR DESCRIPTION
**Pull request by nak3**
This patch adds TestMultipleHosts and TestTimeout.

/cc @ZhiminXiang @tcnghia @markusthoemmes

### CODE REVIEW

File: test/conformance/ingressv2/util.go [hunk 1]
```diff
@@ -308,7 +308,7 @@ func CreateTimeoutService(ctx context.Context, t *testing.T, clients *test.Clien
 		},
 	}
 
-	return name, port, createPodAndService(ctx, t, clients, pod, svc)
+	return name, gatewayv1alpha1.PortNumber(port), createPodAndService(ctx, t, clients, pod, svc)
```
Comment (markusthoemmes) [inline test/conformance/ingressv2/util.go:311]: Same comment as in the other PR: Should we do this at the call-site? Is it valuable to keep these functions generic between our tests?

**Review by markusthoemmes — DISMISSED** [review]
/lgtm
/approve
/hold

For the nit

File: test/conformance/ingressv2/util.go (hunk 1 above)
Comment (nak3) [reply test/conformance/ingressv2/util.go]: Yes, I wondered this. Updated the util function.

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

**Pull request merged**